qcow.c

```diff
@@ -279,16 +279,29 @@ static int qcow_probe(struct bdev *bdev, int dirfd, const char *pathname)
 
 	tcmu_dbg("%s\n", __func__);
 
-	if (faccessat(dirfd, pathname, R_OK|W_OK, AT_EACCESS) == -1)
+	if (faccessat(dirfd, pathname, R_OK|W_OK, AT_EACCESS) == -1) {
+		tcmu_err("faccessat failed dirfd %d, pathname %s, errno %d\n",
+			 dirfd, pathname, errno);
 		return -1;
-	if ((fd = openat(dirfd, pathname, O_RDONLY)) == -1)
+	}
+	if ((fd = openat(dirfd, pathname, O_RDONLY)) == -1) {
+		tcmu_err("openat failed dirfd %d, pathname %s, errno %d\n",
+			 dirfd, pathname, errno);
 		return -1;
-	if (pread(fd, &head, sizeof(head), 0) == -1)
+	}
+	if (pread(fd, &head, sizeof(head), 0) == -1) {
+		tcmu_err("read failed fd: %d\n", fd);
 		goto err;
-	if (be32toh(head.magic) != QCOW_MAGIC)
+	}
+	if (be32toh(head.magic) != QCOW_MAGIC) {
+		tcmu_warn("not qcow: will treat as raw: %s", pathname);
 		goto err;
-	if (be32toh(head.version) != 1)
+	}
+	if (be32toh(head.version) != 1) {
+		tcmu_err("version = %d\n", head.version);
```

Comment:
If the file is qcow2, this will show up as an error in the log, which is misleading.